### ISSUE TITLE
broadcast between SMatrix{M,N} and SMatrix{M,1} (or SMatrix{1,N}) doesn't work

### ISSUE THREAD

**Issue by wsshin** [JuliaArrays/StaticArrays.jl#242]
Broadcasting an operator over a matrix (`SMatrix{M,N}`) and a column-like (`SMatrix{M,1}`) or row-like (`SMatrix{1,N}`) matrix doesn't work and generates a `DimensionMismatch` error is generated.  (Seems to be related to https://github.com/JuliaArrays/StaticArrays.jl/issues/199.)
```julia
julia> VERSION
v"0.6.1-pre.0"

julia> SA = @SMatrix [1 2; 3 4]
2×2 StaticArrays.SArray{Tuple{2,2},Int64,2,4}:
 1  2
 3  4

julia> SC = @SMatrix [2;4]
2×1 StaticArrays.SArray{Tuple{2,1},Int64,2,2}:
 2
 4

julia> SR = @SMatrix [2 4]
1×2 StaticArrays.SArray{Tuple{1,2},Int64,2,2}:
 2  4

julia> broadcast(+, SA, SC)
ERROR: DimensionMismatch("Tried to broadcast on inputs sized Tuple{Int64,Int64}[(2, 2), (2, 1)]")
Stacktrace:
 [1] _broadcast(...) at /Users/wsshin/.julia/v0.6/StaticArrays/src/broadcast.jl:52
 [2] broadcast(::Function, ::StaticArrays.SArray{Tuple{2,2},Int64,2,4}, ::StaticArrays.SArray{Tuple{2,1},Int64,2,2}) at /Users/wsshin/.julia/v0.6/StaticArrays/src/broadcast.jl:8

julia> broadcast(+, SA, SR)
ERROR: DimensionMismatch("Tried to broadcast on inputs sized Tuple{Int64,Int64}[(2, 2), (1, 2)]")
Stacktrace:
 [1] _broadcast(...) at /Users/wsshin/.julia/v0.6/StaticArrays/src/broadcast.jl:52
 [2] broadcast(::Function, ::StaticArrays.SArray{Tuple{2,2},Int64,2,4}, ::StaticArrays.SArray{Tuple{1,2},Int64,2,2}) at /Users/wsshin/.julia/v0.6/StaticArrays/src/broadcast.jl:8
```
Is this difficult to fix?

If an actual vector is used instead of a column-like matrix, broadcasting works.  Therefore, currently I am working around this as follows:
```julia
julia> broadcast(+, SA, SC[:,1])
2×2 StaticArrays.SArray{Tuple{2,2},Int64,2,4}:
 3  4
 7  8

julia> broadcast(+, SA', SR[1,:])'
2×2 StaticArrays.SArray{Tuple{2,2},Int64,2,4}:
 3  6
 5  8
```

**Comment by andyferris** [comment]
We definitely need to fix this...

**Comment by wsshin** [comment]
Closed via #274.
